astropy/io/fits/tests/test_image.py

```diff
@@ -1725,3 +1725,12 @@ def test_bzero_mishandled_info(tmpdir):
     hdu.writeto(filename, clobber=True)
     hdul = fits.open(filename)
     hdul.info()
+
+
+def test_image_write_readonly(tmpdir):
+    # Regression test to make sure that we can write out read-only arrays (#5512)
+    x = np.array([1,2,3])
+    x.setflags(write=False)
+    ghdu = fits.ImageHDU(data=x)
+    ghdu.add_datasum()
+    ghdu.writeto(tmpdir.join('test.fits').strpath)
```

Comment:
I tried writing a regression test for GroupsHDU, but I couldn't get it to go through the read-only logic I added - I just put it there in case but I'm not sure how to test it. But yes I should modify this test to read in the data and check the values.